### PR TITLE
GRPC V2 : Get Block Item Status

### PR DESCRIPTION
**Pull request by tds-1**
## Purpose

Added support for GRPC V2 GetBlockItemInfo

## Changes

Added support for GRPC V2 GetBlockItemInfo

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

_Remove if not applicable.

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

### CODE REVIEW

File: concordium-sdk/src/main/java/com/concordium/sdk/ClientV2MapperExtensions.java [hunk 1]
```diff
@@ -763,4 +771,319 @@ static ConsensusStatus to(ConsensusInfo concensusInfo) {
 
         return builder.build();
     }
+
+
+    // Convert a com.concordium.grpc.v2.BlockItemStatus object to the corresponding com.concordium.sdk.responses.TransactionStatus object
+    static TransactionStatus to(com.concordium.grpc.v2.BlockItemStatus blockItemStatus) {
+        var builder = TransactionStatus.builder();
+        if (blockItemStatus.getStatusCase() == BlockItemStatus.StatusCase.FINALIZED){
```
Comment (parv0888) [inline concordium-sdk/src/main/java/com/concordium/sdk/ClientV2MapperExtensions.java:779]: can be converted to switch

File: concordium-sdk/src/main/java/com/concordium/sdk/ClientV2MapperExtensions.java [hunk 2]
```diff
@@ -763,4 +771,319 @@ static ConsensusStatus to(ConsensusInfo concensusInfo) {
...
+            builder = builder.status(Status.FINALIZED)
+                    .outcomes(to(blockItemStatus.getFinalized().getOutcome()));
+
+        }
+        else if (blockItemStatus.getStatusCase() == BlockItemStatus.StatusCase.COMMITTED){
```
Comment (parv0888) [inline concordium-sdk/src/main/java/com/concordium/sdk/ClientV2MapperExtensions.java:784]: can be converted to switch

File: concordium-sdk/src/main/java/com/concordium/sdk/ClientV2MapperExtensions.java [hunk 3]
```diff
@@ -763,4 +771,319 @@ static ConsensusStatus to(ConsensusInfo concensusInfo) {
...
+            builder = builder.status(Status.COMMITTED)
+                    .outcomes(to(blockItemStatus.getCommitted().getOutcomesList()));
+        }
+        else if (blockItemStatus.getStatusCase() == BlockItemStatus.StatusCase.RECEIVED){
```
Comment (parv0888) [inline concordium-sdk/src/main/java/com/concordium/sdk/ClientV2MapperExtensions.java:788]: can be converted to switch

**Review by parv0888 — APPROVED** [review]

File: concordium-sdk/src/main/java/com/concordium/sdk/ClientV2MapperExtensions.java (hunk 1 above)
Comment (tds-1) [reply concordium-sdk/src/main/java/com/concordium/sdk/ClientV2MapperExtensions.java]: Done

File: concordium-sdk/src/main/java/com/concordium/sdk/ClientV2MapperExtensions.java (hunk 2 above)
Comment (tds-1) [reply concordium-sdk/src/main/java/com/concordium/sdk/ClientV2MapperExtensions.java]: Done

File: concordium-sdk/src/main/java/com/concordium/sdk/ClientV2MapperExtensions.java (hunk 3 above)
Comment (tds-1) [reply concordium-sdk/src/main/java/com/concordium/sdk/ClientV2MapperExtensions.java]: Done

File: concordium-sdk/src/test/java/com/concordium/sdk/ClientV2GetBlockItemStatusTest.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+public class ClientV2GetBlockItemStatusTest {
+
+    private static final int TRANSACTION_INDEX = 5;
+    private static final int ENERGY_COST = 500000;
+    private static final String SENDER_ADDRESS = "48x2Uo8xCMMxwGuSQnwbqjzKtVqK5MaUud4vG7QEUgDmYkV85e";
+    private static final String BLOCK_HASH = "d1bf95c1a2acc0947ec3900040c2ba172071aa759adf269c55ebb896aa6825c2";
+    private static final String TRANSACTION_HASH = "1ea074f0e12e18684f2d6bbf2039c6db32d2fd5c28e6ba74c8e92f36e88b1901";
+    private static final long TRANSACTION_COST = 123121;
+
+    private static final BlockItemStatus GRPC_BLOCK_ITEM_STATUS = BlockItemStatus.newBuilder()
```
Comment (EmilLa1) [inline concordium-sdk/src/test/java/com/concordium/sdk/ClientV2GetBlockItemStatusTest.java:39]: Indentation looks a bit off here?

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/transactionstatus/DelegationStakeDecreased.java [hunk 5]
```diff
@@ -22,6 +22,6 @@ public class DelegationStakeDecreased extends AbstractDelegatorResult {
 
     @Override
     public TransactionResultEventType getType() {
-        return TransactionResultEventType.BAKER_STAKE_DECREASED;
+        return TransactionResultEventType.DELEGATION_STAKE_DECREASED;
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/transactionstatus/DelegationStakeDecreased.java:25]: Please add this bugfix to the changelog. Something like: Fixed incorrect transaction event with respect to a transaction status for when a delegator decreased its stake.

**Review by EmilLa1 — COMMENTED** [review]
Looks good! GJ. 

Please also add the bug fix to the changelog.

**Review by EmilLa1 — APPROVED** [review]
Looks good. Please merge with main.

**Pull request merged**